components/gitpod-cli/cmd/idp-login-aws.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	idpAudienceAWS = "sts.amazonaws.com"
+)
+
+var idpLoginAwsOpts struct {
+	RoleARN         string
+	CredentialsFile string
+}
+
+var idpLoginAwsCmd = &cobra.Command{
+	Use:   "aws",
+	Short: "Login to AWS",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
```

Comment:
Check if the aws binary exists to avoid confusing errors?

Reply:
This is the behaviour if the AWS binary is not available:
<img width="365" alt="image" src="https://user-images.githubusercontent.com/3210701/222718490-92c2e68f-399f-4852-9752-01987fe01ca3.png">

How should the error look like instead?

Reply:
I was thinking of requesting installing the CLI, but that is just fine.

Reply:
We could link to the [AWS Docs](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html). Arguably though, if someone's using this feature they're deep enough into it that they'll have looked at our (yet to be written) docs which will need to state that the AWS CLI needs to be installed.